src/python/pants/engine/fs.py

```diff
@@ -29,24 +30,47 @@ class Path(datatype(['path', 'stat'])):
   """
 
 
-class PathGlobs(datatype(['include', 'exclude'])):
+class PathGlobs(datatype([
+    'include',
```

Comment:
I'd be tempted to add `tuple` type constraints to `include` and `exclude` - having half-typed objects seems weird